platforms/jvm/war/src/main/java/org/gradle/api/plugins/internal/DefaultWarPluginConvention.java

```diff
@@ -27,7 +26,8 @@
 
 import static org.gradle.api.reflect.TypeOf.typeOf;
 
-public abstract class DefaultWarPluginConvention extends WarPluginConvention implements HasPublicType {
+@SuppressWarnings("deprecation")
```

Comment:
Nit: instead of suppressing deprecations in this class, IMO we should mark it deprecated

Reply:
Actually this won't completely work, because of the extension.  Have to suppress the usage at least.